### PR TITLE
F calibrate

### PR DESCRIPTION
**Pull request by kirstyellis**

### CODE REVIEW

File: src/polhemus.cpp [hunk 1]
```diff
@@ -39,68 +41,84 @@ int Polhemus::count_bits(uint16_t v) {
 int Polhemus::device_write(uint8_t *buf, int size, int timeout)
 {
   int nActual = 0;
-  int retval = 0;
+  int retval = RETURN_ERROR;
 
   retval = libusb_bulk_transfer(device_handle, endpoint_out, buf, size, &nActual, timeout);
 
-  if (retval)
+  for (int i=0; i < nActual; i++)
```
Comment (giusebar) [inline src/polhemus.cpp:48]: do we need this??

File: src/polhemus.cpp [hunk 2]
```diff
@@ -109,23 +127,225 @@ int Polhemus::device_read(void *pbuf, int &size, bool bTOisErr)
...
+        {
+          retval = 0;
+        }
+        if (!retval)
+        {
+          tf2::Quaternion q = get_quaternion(i);
+          double roll, pitch, yaw;
+          tf2::Matrix3x3(q).getRPY(roll, pitch, yaw);
+
+//          fprintf(stderr, "quat x: %f\n", q[0]);
```
Comment (giusebar) [inline src/polhemus.cpp:195]: remove comments?

File: src/polhemus.cpp [hunk 3]
```diff
@@ -39,68 +41,84 @@ int Polhemus::count_bits(uint16_t v) {
...
-  int retval = 0;
+  uint32_t timeout = VPUSB_READ_TIMEOUT_MS;
+  int retval = RETURN_ERROR;
   int nActual = 0;
   unsigned char *pbuf_c = (unsigned char*) pbuf;
+
   retval = libusb_bulk_transfer(device_handle, endpoint_in, pbuf_c, size, &nActual, timeout);
 
-  if ((retval == LIBUSB_ERROR_TIMEOUT))
+  for (int i=0; i < nActual; i++)
```
Comment (giusebar) [inline src/polhemus.cpp:108]: we should maybe have a flag instead of for loops?

File: src/polhemus.cpp [hunk 4]
```diff
@@ -109,23 +127,225 @@ int Polhemus::device_read(void *pbuf, int &size, bool bTOisErr)
...
+  }
+
+  for (int i = 0; i < station_count; ++i)
+  {
+    tf2::Quaternion q = get_quaternion(i);
+
+    double roll, pitch, yaw;
+    tf2::Matrix3x3(q).getRPY(roll, pitch, yaw);
+
+//    fprintf(stderr, "quat x: %f\n", q[0]);
```
Comment (giusebar) [inline src/polhemus.cpp:276]: remove comments

File: src/polhemus.cpp [hunk 5]
```diff
@@ -109,23 +127,225 @@ int Polhemus::device_read(void *pbuf, int &size, bool bTOisErr)
...
+//    fprintf(stderr, "quat y: %f\n", q[1]);
+//    fprintf(stderr, "quat z: %f\n", q[2]);
+//    fprintf(stderr, "quat w: %f\n", q[3]);
+
+    // convert to degrees
+    roll = (roll * 180) / 3.14;
+    pitch = (pitch * 180) / 3.14;
+    yaw = (yaw * 180) / 3.14;
+
+//    fprintf(stderr, "roll: %f\n", roll);
```
Comment (giusebar) [inline src/polhemus.cpp:286]: remove comments

File: src/polhemus.cpp (hunk 1 above)
Comment (kirstyellis) [reply src/polhemus.cpp]: done

File: src/polhemus.cpp (hunk 2 above)
Comment (kirstyellis) [reply src/polhemus.cpp]: done

File: src/polhemus.cpp (hunk 3 above)
Comment (kirstyellis) [reply src/polhemus.cpp]: done

File: src/polhemus.cpp (hunk 4 above)
Comment (kirstyellis) [reply src/polhemus.cpp]: done

File: src/polhemus.cpp (hunk 5 above)
Comment (kirstyellis) [reply src/polhemus.cpp]: done

**Pull request merged**